awx/ui_next/src/api/models/Credentials.js

```diff
@@ -15,10 +15,34 @@ class Credentials extends Base {
     });
   }
 
-  readInputSources(id, params) {
-    return this.http.get(`${this.baseUrl}${id}/input_sources/`, {
-      params,
-    });
+  readInputSources(id) {
+    const maxRequests = 5;
+    let requestCounter = 0;
+    const fetchInputSources = async (pageNo = 1, inputSources = []) => {
+      try {
+        requestCounter++;
+        const { data } = await this.http.get(
+          `${this.baseUrl}${id}/input_sources/`,
+          {
+            params: {
+              page: pageNo,
+              page_size: 200,
+            },
+          }
+        );
+        if (data.next && requestCounter <= maxRequests) {
+          return fetchInputSources(
+            pageNo + 1,
+            inputSources.concat(data.results)
+          );
+        }
+        return Promise.resolve(inputSources.concat(data.results));
```

Comment:
The credential edit component expects this return structure to be `{ data: { results: [...] } }`.

Reply:
https://github.com/ansible/awx/pull/7281/commits/bbafeadd0feeecf393b9ccc719485d01d242ef9d